crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3431,6 +3452,139 @@ impl<'db> TypeInferenceBuilder<'db> {
+                                "Type arguments for `Literal` must be None, a literal value (int, bool, str, or bytes), or an enum value",
+                            ),
+                        );
+                        return Type::Unknown;
+                    }
+                };
+
+                let slice_ty = self.infer_literal_parameter_type(parameters);
+
+                match slice_ty {
+                    Type::Never
+                    | Type::Unknown
+                    | Type::Unbound
+                    | Type::Todo
+                    | Type::FunctionLiteral(_)
+                    | Type::ModuleLiteral(_)
+                    | Type::ClassLiteral(_)
+                    | Type::Union(_)
+                    | Type::Intersection(_)
+                    | Type::Any => {
+                        self.add_diagnostic(
+                                    parameters.into(),
+                                    "invalid-literal-parameter",
+                                    format_args!(
+                                        "Type arguments for `Literal` must be None, a literal value (int, bool, str, or bytes), or an enum value",
+                                    ),
+                                );
+                        Type::Unknown
+                    }
+                    Type::Tuple(tuple) => {
+                        let elts = tuple.elements(self.db);
+                        Type::Union(UnionType::new(self.db, elts))
+                    }
+                    ty => ty,
+                }
+            }
+        }
+    }
+
+    fn infer_literal_parameter_type(&mut self, parameters: &ast::Expr) -> Type<'db> {
+        // If the slice itself is another subscript and it is another Literal then flatten
+        // the values
+        match parameters {
+            ruff_python_ast::Expr::Subscript(inner_literal_subscript) => {
+                let inner_subscript_value = self.infer_expression(&inner_literal_subscript.value);
+                if let Type::Instance(inner_instance) = inner_subscript_value {
+                    if inner_instance.is_known(KnownInstance::Literal) {
+                        self.infer_parameterized_known_instance_type_expression(
+                            KnownInstance::Literal,
+                            &inner_literal_subscript.slice,
+                        )
+                    } else {
+                        Type::Unknown
+                    }
+                } else {
+                    self.add_diagnostic(
+                                        parameters.into(),
+                                        "invalid-literal-parameter",
+                                        format_args!(
+                                            "Type arguments for `Literal` must be None, a literal value (int, bool, str, or bytes), or an enum value",
+                                        ),
+                                    );
```

Comment:
Thanks I was confused why it's not being formatted. I cannot apply this because I resolved other comments first but I will double check final changes are correctly formatted.